packages/idyll-components/src/dynamic.js

```diff
@@ -25,18 +25,34 @@ class Dynamic extends React.PureComponent {
     this.drag(Selection.select(node));
   }
 
-  render() {
-    const { format, value } = this.props;
+  transformValue() {
+    const { format, value, map, min, step } = this.props;
+    const mapOnArray = Array.isArray(map);
+    const mapOnConditions = typeof map === 'string';
+    if (mapOnArray) {
+      const commonDiff = Math.ceil(Math.abs(step));
+      const index = Math.floor((value - min) / commonDiff);
+      return index < map.length ? map[index] : value;
+    }
```

Comment:
I'm worried this mapping might be a bit too "magical". 

I think it may be more straightforward to accept a function that users can define to specify there own mapping, e.g. 

```
[Dynamic value:x map:`(d) => { return ... }`]
```

Reply:
I have updated this so now the format prop accepts a function.